From 555504166852a9b9c56ac4e1fb5bb1bf20cbf8ad Mon Sep 17 00:00:00 2001
From: Bobby Bingham <koorogi@koorogi.info>
Date: Sun, 26 Mar 2017 14:50:37 -0500
Subject: s390x: provide sigcontext struct definition

This structure was missed when creating the s390x port.

This is based on the report and patch from William Pitcock, but with a
modified structure defintion to more closely match the kernel's
definition.
---
 arch/s390x/bits/signal.h | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/arch/s390x/bits/signal.h b/arch/s390x/bits/signal.h
index c866583..e5aca4b 100644
--- a/arch/s390x/bits/signal.h
+++ b/arch/s390x/bits/signal.h
@@ -33,6 +33,21 @@ typedef struct
 	fpregset_t fpregs;
 } mcontext_t;
 
+struct sigcontext {
+	unsigned long oldmask[1];
+	struct {
+		struct {
+			__psw_t psw;
+			unsigned long gprs[16];
+			unsigned acrs[16];
+		} regs;
+		struct {
+			unsigned fpc;
+			double fprs[16];
+		} fpregs;
+	} *sregs;
+};
+
 #else
 
 typedef struct {
-- 
cgit v0.11.2

